### DISCUSSION TITLE
Bring images with the research results

### DISCUSSION THREAD

**Discussion by pedrorocha-net** [assafelovic/gpt-researcher#822] [Ideas]
The generated reports are already highly valuable, so I was thinking what if we could also bring images related to it? It's similar to Perplexity.ai.

Most of the Search APIs like Serper.dev already can provide images in their JSON, but we would have to think about how to handle the images, to "select" them. Maybe we could simply return the list of images and allow devs using GPTR as a pip package to get it and manipulate as they want.

Any thoughts about that?

**Comment by ElishaKay** [comment]
Fair point - could add a lot of color & variation

**Comment by bhavesh-vwaves** [comment]
Has there been any progress or updates on this feature? Incorporating relevant images could significantly enhance the quality and engagement of the reports.

**Reply by ElishaKay** [reply]
welcome @bhavesh-bootnext 

yes - images are available in the UX - [you have the demo here](https://github.com/assafelovic/gpt-researcher)

**Reply by bhavesh-vwaves** [reply]
Thank you!

I'm currently utilizing GPT Researcher and appreciate its capabilities. I noticed that while it supports exporting reports to formats like PDF and Word, there isn't a direct option for exporting reports in HTML format. Is there a way to enable or add support for HTML export? This feature would be highly beneficial for integrating reports into web-based platforms.

**Reply by ElishaKay** [reply]
Fair point - feel free to go for the PR, if you'd like - we should probably convert the "Export to" buttons into a dropdown with the relevant options

The new NextJS frontend saves the reports as localstorage on the user's browser - pretty solid
